website/content/plugins/drivers/community/pledge.mdx

```diff
@@ -0,0 +1,333 @@
+
+## Resource Isolation
+
+CPU and memory resource isolation is enforced by cgroups v2 features, and is
+configured through the resources block in task configuration.
+
+```hcl
+resources {
+  cores  = 3
+  memory = 512
+}
+```
+
+#### cpu bandwidth
+
+The `pledge` driver always enforces a maximum CPU bandwidth available to the
+task. The bandwidth is calculated whether `resources.cpu` or `resources.cores`
+is configured.
+
+If `resources.cores` is set, the scheduler reserves that many CPU cores for use
+by the task, and only this task will be able to run on those cores.
+
+#### memory oversubscription
+
+If the Nomad scheduler is configured to enable memory [oversubscription]
+[oversub], the `pledge` driver will correctly enable configuring `memory_max`
+in addition to `memory`. In this case, `memory_max` indicates the maximum amount
+of memory the task is able to request before being OOM killed, and `memory`
+represents a minimum amount of memory the kernel will gauruntee is available for
+the Task.
+
+```hcl
+resources {
+  cpu = 2000
+  memory = 1024
+  memory_max = 2048
+}
+```
+
+## Capabilities
+
+The `pledge` driver implements the following [capabilities][capabilities].
+
+| Feature              | Implementation    |
+| ------------         | ------------------|
+| send signals         | true              |
+| exec                 | false             |
+| filesystem isolation | Landlock LSM      |
+| network isolation    | host, group, none |
+| volume mounting      | false             |
+
+For sending signals, use the `nomad alloc signal` command.
+
+## Examples
+
+A larger set of examples are available in the [source][hack] repository.
+
+Showing how to `curl example.com`.
+
+```hcl
+job "curl" {
+  type = "batch"
+  group "group" {
+    task "curl" {
+      driver = "pledge"
+      user   = "nobody"
+      config {
+        command  = "curl"
+        args     = ["example.com"]
+        promises = "stdio rpath inet dns sendfd"
+      }
+    }
+  }
+}
+```
+
+An example using the Python built-in HTTP server to render a trivial web page.
+
+```hcl
+job "http" {
+  group "group" {
+    network {
+      mode = "host"
+      port "http" { static = 8181 }
+    }
+
+    task "task" {
+      driver = "pledge"
+      user   = "nobody"
+      config {
+        command    = "python3"
+        args       = ["-m", "http.server", "${NOMAD_PORT_http}", "--directory", "${NOMAD_TASK_DIR}"]
+        promises   = "stdio rpath inet"
+        unveil     = ["r:/etc/mime.types", "r:${NOMAD_TASK_DIR}"]
+      }
+
+      template {
+        destination = "local/index.html"
+        data        = <<EOH
+<!doctype html>
+<html>
+  <title>example</title>
+  <body><p>Hello, friend!</p></body>
+</html>
+EOH
+      }
+    }
+  }
+}
+```
+
+## Troubleshooting
+
+When setting up a new Task using the `pledge` task driver, it helps to run
+the command manually using the `pledge.com` utility to make sure the necessary
```

Comment:
The text is inconsistent on whether it uses "pledge.com" or "`pledge.com`". Maybe we should use the backticks everywhere? (This also prevents folks from thinking it's a domain name.)

Reply:
The tool itself is actually being renamed to `pledge`

https://github.com/jart/pledge/commit/f0e63fecb8d68a8aa9f4b15e5e34b31640080f7e

but hasn't been released yet with that name